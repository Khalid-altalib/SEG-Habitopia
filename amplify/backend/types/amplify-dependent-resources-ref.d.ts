export type AmplifyDependentResourcesAttributes = {
  "api": {
    "Habitopia": {
      "GraphQLAPIEndpointOutput": "string",
      "GraphQLAPIIdOutput": "string",
      "GraphQLAPIKeyOutput": "string"
    }
  },
  "auth": {
    "Habitopia": {
      "AppClientID": "string",
      "AppClientIDWeb": "string",
      "IdentityPoolId": "string",
      "IdentityPoolName": "string",
      "UserPoolArn": "string",
      "UserPoolId": "string",
      "UserPoolName": "string"
    }
  }
}