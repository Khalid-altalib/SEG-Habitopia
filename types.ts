import type { NativeStackNavigationProp } from "@react-navigation/native-stack";


export type RootParams = {
  Auth: undefined,
  Existing: undefined,
  Chat : undefined,
}

export type ProfileParams = {
  Home: undefined;
  Chats: undefined;
  Welcome: undefined;
  Explore: undefined;
  Leaderboard: undefined;
  You: undefined;
  Chat: undefined;
}

export type AuthParams = {
  Welcome: undefined;
  SignUp: undefined;
  LogIn: undefined;
};

export type ChatParams= {
  Chat: undefined;
}


export type LocalUser = {
  authToken: string;
  username: string;
};

