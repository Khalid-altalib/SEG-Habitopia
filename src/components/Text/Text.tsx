// React
import React, { useEffect, useState } from "react";

// React Native
import { StyleSheet, Text as RnText } from "react-native";

// Native Base
import { theme, useColorModeValue } from "native-base";

// Expo Font
import * as Font from "expo-font";

// Expo Splash Screen
import * as SplashScreen from "expo-splash-screen";

// Habitopia
import { TextType } from "types";

import { act } from "@testing-library/react-native";

export type TextProps = {
  /** The type of text. Affects font family, weight, etc. */
  type?: TextType;
  /** Any additional styles to apply to the text. */
  style?: any;
  /** The actual text itself, along with possibly an icon. */
  children: any;
  /** The color of the text. */
  color?: string;
};

const Text = ({ type, style, children, color }: TextProps) => {
  const [fontsLoaded, setFontsLoaded] = useState(false);

  const loadFonts = async () => {
    try {
      await Font.loadAsync({
        "Roboto Black": require("../../../assets/fonts/Roboto/Black.ttf"),
        "Roboto Medium": require("../../../assets/fonts/Roboto/Medium.ttf"),
      });
    } catch (error) {
      console.log(error);
    }

    act(() => {
      setFontsLoaded(true);
    });
  };

  useEffect(() => {
    async function loading() {
      await loadFonts();
      act(() => {
        setFontsLoaded(true);
      });
    }
    loading();
  }, []);

  useEffect(() => {
    async function hideSplashScreen() {
      if (fontsLoaded) {
        try {
          await SplashScreen.hideAsync();
        } catch (error) {
          console.log(error);
        }
      }
    }
    hideSplashScreen();
  }, [fontsLoaded]);

  if (type == undefined) type = TextType.Regular;

  const additionalStyle = TextType[type]
    .toString()
    .toLowerCase() as keyof typeof styles;

  const styles = StyleSheet.create({
    base: {
      color:
        color ||
        useColorModeValue(theme.colors.gray[900], theme.colors.gray[200]),
    },
    regular: {
      fontSize: 19,
      fontFamily: fontsLoaded ? "Roboto Medium" : undefined,
    },
    small: {
      fontSize: 15,
      fontFamily: fontsLoaded ? "Roboto Medium" : undefined,
    },
    subtle: {
      fontSize: 15,
      color:
        color ||
        useColorModeValue(theme.colors.gray[500], theme.colors.gray[400]),
      fontFamily: fontsLoaded ? "Roboto Medium" : undefined,
    },
    heading: {
      fontSize: 40,
      fontFamily: fontsLoaded ? "Roboto Black" : undefined,
    },
    subheading: {
      fontSize: 23,
      fontFamily: fontsLoaded ? "Roboto Black" : undefined,
    },
    button: {
      color: useColorModeValue(theme.colors.gray[200], theme.colors.gray[200]),
      fontSize: 19,
      fontFamily: fontsLoaded ? "Roboto Medium" : undefined,
    },
  });

  return (
    <RnText style={[styles.base, styles[additionalStyle], style]}>
      {children}
    </RnText>
  );
};

export default Text;
