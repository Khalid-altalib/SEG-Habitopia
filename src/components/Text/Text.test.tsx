// React
import React from "react";

// React Native Testing Library
import { render, screen } from "@testing-library/react-native";

// Habitopia
import Text from "@components/Text";
import { TextType } from "types";

it("Renders correctly.", () => {
  render(<Text type={TextType.Regular}>This is some example content.</Text>);

  expect(screen.toJSON()).toMatchSnapshot();
});
