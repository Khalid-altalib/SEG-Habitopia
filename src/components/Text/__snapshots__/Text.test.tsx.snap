// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly. 1`] = `
<Text
  style={
    [
      {
        "color": "#18181b",
      },
      {
        "fontFamily": undefined,
        "fontSize": 19,
      },
      undefined,
    ]
  }
>
  This is some example content.
</Text>
`;
